h2o-algos/src/main/java/hex/mli/loco/LeaveOneCovarOut.java

```diff
@@ -0,0 +1,245 @@
+        Vec[] _result;
+
+        public LeaveOneCovariateOutDriver(Frame locoFrame, Frame fr, Model m, String predictor, String replaceVal){
+            _locoFrame = locoFrame;
+            _frame = fr;
+            _model = m;
+            _predictor = predictor;
+            _replaceVal = replaceVal;
+        }
+
+        @Override
+        public void compute2() {
+            if(_model._output.getModelCategory() == ModelCategory.Multinomial){
+                Vec[] predTmp = getNewPredictions(_model,_frame,_predictor,_replaceVal);
+                Frame tmpFrame = new Frame().add(_locoFrame).add(new Frame(predTmp));
+                _result = new MultiDiffTask(_model._output.nclasses()).doAll(Vec.T_NUM, tmpFrame).outputFrame().vecs();
+                for (Vec v : predTmp) v.remove();
+            } else {
+                _result = getNewPredictions(_model, _frame, _predictor,_replaceVal);
+                new DiffTask().doAll(_locoFrame.vec(0), _result[0]);
+            }
+            Log.info("Completed Leave One Covariate Out (LOCO) Analysis for column: " + _predictor);
+            tryComplete();
+        }
+
+    }
+
+    /**
+     * Get base predictions given a model and frame. "Base predictions" are predictions based on all features in the
+     * model
+     * @param m An H2O supervised model
+     * @param fr A Frame to score on
+     * @return An array of Vecs containing predictions
+     */
+    private static Vec[] getBasepredictions(Model m, Frame fr){
+
+        Frame basePredsFr = m.score(fr,null,null,false);
+
+        if(m._output.getModelCategory() == ModelCategory.Binomial) {
+            Vec basePreds = basePredsFr.remove(2);
+            basePredsFr.delete();
+            return new Vec[] {basePreds};
+        }else if(m._output.getModelCategory() == ModelCategory.Multinomial){
+            Vec[] basePredsVecs = basePredsFr.vecs();
+            DKV.remove(basePredsFr._key);
+            return basePredsVecs;
+        } else {
+            Vec basePreds = basePredsFr.remove(0);
+            basePredsFr.delete();
+            return new Vec[] {basePreds};
+        }
+
+    }
+
+    /**
+     * Get new predictions based on dropping/replacing a column with mean, median, or mode
+     * @param m An H2O supervised model
+     * @param fr A Frame to score on
+     * @param colToDrop Column to modify/drop before prediction
+     * @param valToReplace Value to replace colToDrop by (Default is null)
+     * @return
+     */
+    private static Vec[] getNewPredictions(Model m, Frame fr, String colToDrop, String valToReplace) {
+        Frame workerFrame = new Frame(fr);
+        Vec vecToReplace = fr.vec(colToDrop);
+        Vec replacementVec = null;
+        if(valToReplace == null){
+            replacementVec = vecToReplace.makeCon(Double.NaN);
+        } else if(valToReplace.equals("mean")){
+            if(vecToReplace.isCategorical()){
+                Vec tmpVec = vecToReplace.makeCon(vecToReplace.mode());
```

Comment:
There is already method `makeCon(final double d, byte type)` - you do not need to create another temporary vector.